dvc/remote/base.py

```diff
@@ -202,6 +209,7 @@ def get_dir_checksum(self, path_info):
         checksum, tmp_info = self._get_dir_info_checksum(dir_info)
         new_info = self.cache.checksum_to_path_info(checksum)
         if self.cache.changed_cache_file(checksum):
+            self.cache.makedirs(new_info.parent)
```

Comment:
this implies that `cache` is `local`, right? are there any downsides of doing this? like a remote cache with `ssh` for example

Reply:
Not really, if remote supports dirs then it should support `makedirs`. So currently ssh and local support makedirs.